### PR TITLE
87 mbgl timeout

### PR DESCRIPTION
**Pull request by gdey**
This PR adds a `timeout` (in minutes) command line parameter to ensure that a job does not run for more than the time out.

The worker will respond to soft kills, enabling it to report back a failure for the job.

### CODE REVIEW

**Comment by gdey** [comment]
fixes #87

**Comment by gdey** [comment]
For the review you will probably want to start with:
* 208196a -- is in the main bit of code that sets up the 
ctx and the timeout
* 8082f05 -- added the SigTerm handling which is built on top of the ctx.

File: cmd/atlante/cmd/root.go [hunk 1]
```diff
@@ -131,10 +146,28 @@ func rootCmdRun(cmd *cobra.Command, args []string) error {
 			ExitCode:  1,
 		}
 	}
-	ctx, cancel := context.WithCancel(context.Background())
+	if timeout != 0 {
```
Comment (ARolek) [inline cmd/atlante/cmd/root.go]: Remove the var declarations at 133, drop the else statement at 153 and use line 154 before the if on 149

File: mbgl/image/image.go [hunk 2]
```diff
@@ -76,78 +80,99 @@ type Image struct {
...
-	im.initLck.Lock()
-	defer im.initLck.Unlock()
-	if err := im.backingStore.Close(); err != nil {
-		log.Printf("warning failed to close %v : %v", im.backingStore.Name(), err)
+	if err := img.backingStore.Close(); err != nil {
+		log.Infof("warning failed to close %v : %v", img.backingStore.Name(), err)
 	}
-	log.Printf("removing backing store %v", im.backingStore.Name())
+	log.Infof("removing backing store %v", img.backingStore.Name())
 	// ignore any errors.
```
Comment (ARolek) [inline mbgl/image/image.go]: leftover comment.

**Pull request merged**